src/pgrepr/src/types.rs

```diff
@@ -551,6 +556,36 @@ impl Type {
             postgres_types::Type::REGTYPE_ARRAY => Type::Array(Box::new(Type::RegType)),
             postgres_types::Type::INT2_VECTOR => Type::Int2Vector,
             postgres_types::Type::INT2_VECTOR_ARRAY => Type::Array(Box::new(Type::Int2Vector)),
+            postgres_types::Type::INT4_RANGE => Type::Range {
+                element_type: Box::new(Type::Int4),
+            },
+            postgres_types::Type::INT4_RANGE_ARRAY => Type::Array(Box::new(Type::Range {
+                element_type: Box::new(Type::Int4),
+            })),
+            postgres_types::Type::NUM_RANGE => Type::Range {
+                element_type: Box::new(Type::Numeric { constraints: None }),
+            },
+            postgres_types::Type::NUM_RANGE_ARRAY => Type::Array(Box::new(Type::Range {
+                element_type: Box::new(Type::Numeric { constraints: None }),
+            })),
+            postgres_types::Type::TS_RANGE => Type::Range {
+                element_type: Box::new(Type::Timestamp { precision: None }),
+            },
+            postgres_types::Type::TS_RANGE_ARRAY => Type::Array(Box::new(Type::Range {
+                element_type: Box::new(Type::Timestamp { precision: None }),
+            })),
+            postgres_types::Type::TSTZ_RANGE => Type::Range {
+                element_type: Box::new(Type::TimestampTz { precision: None }),
+            },
+            postgres_types::Type::TSTZ_RANGE_ARRAY => Type::Array(Box::new(Type::Range {
+                element_type: Box::new(Type::TimestampTz { precision: None }),
+            })),
+            postgres_types::Type::DATE_RANGE => Type::Range {
+                element_type: Box::new(Type::Date),
+            },
+            postgres_types::Type::DATE_RANGE_ARRAY => Type::Array(Box::new(Type::Range {
+                element_type: Box::new(Type::Date),
+            })),
```

Comment:
Probably not necessary, but the only range we're missing that Postgres supports is a `bigint` range.
https://github.com/sfackler/rust-postgres/blob/97db7770783361ae12f2104dc138b84b8c883b66/codegen/src/pg_type.dat#L494-L498

Reply:
Good catch! My intellitype must have missed it.

Reply:
Oops, I meant to link the actual postgres file, not the rust-postgres version: 
https://github.com/postgres/postgres/blob/f03bd5717eaf31569ca797a2f7d65608f88ac2a2/src/include/catalog/pg_type.dat#L494-L498

Either way, they're identical. I just thought I'd still drop the official one in for confirmation.